src/components/common/DataTable.vue

```diff
@@ -79,38 +79,48 @@ export interface Item {
 export default class DataTable extends Vue {
   @Prop() fields!: ViewTableField[];
   @Prop() items!: Item[];
-  @Prop() filter!: string;
-  @Prop() filterBy!: string;
-  @Prop() defaultSort!: string;
+  @Prop() filter?: string;
+  @Prop() filterBy?: string;
+  @Prop() defaultSort?: string;
   @Prop({ default: "desc" }) defaultOrder!: "desc" | "asc";
   @Prop({ default: 10 }) perPage!: number;
   @Prop({ default: false }) hidePagination!: boolean;
   @Prop() filterFunction?: Function;
 
-  sortBy: string = this.defaultSort;
+  sortBy: string = ''
   descOrder: boolean = this.defaultOrder === "desc";
   currentPage = 1;
 
+  created() {
+    this.sortBy = this.defaultSort ? this.defaultSort : '';
+  }
+
   get modifiedItems() {
     let filtered = [];
     const items = this.items;
```

Comment:
Let's add a ```.slice()``` on the end of this, just in case. Not entirely sure if we can mutate this source even by reference or not because I know that's possible in Vue 3.

Reply:
added .slice() as suggested